src/store/poolSlice.ts

```diff
@@ -0,0 +1,434 @@
+      return pool.repayWithPermit({
+        user: currentAccount,
+        reserve: poolAddress,
+        amount: amountToRepay, // amountToRepay.toString(),
+        interestRateMode: debtType,
+        signature,
+        useOptimizedPath: get().useOptimizedPath(),
+        deadline,
+      });
+    },
+    supply: ({ poolAddress, amountToSupply }) => {
+      const pool = getCorrectPool();
+      const currentAccount = get().account;
+      if (pool instanceof Pool) {
+        return pool.supply({
+          user: currentAccount,
+          reserve: poolAddress,
+          amount: amountToSupply,
+          useOptimizedPath: get().useOptimizedPath(),
+        });
+      } else {
+        const lendingPool = pool as LendingPool;
+        return lendingPool.deposit({
+          user: currentAccount,
+          reserve: poolAddress,
+          amount: amountToSupply,
+        });
+      }
+    },
+    supplyWithPermit: (args) => {
+      const pool = getCorrectPool() as Pool;
+      const user = get().account;
+      return pool.supplyWithPermit({
+        ...args,
+        user,
+        useOptimizedPath: get().useOptimizedPath(),
+      });
+    },
+    swapCollateral: async ({
+      poolReserve,
+      targetReserve,
+      priceRoute,
+      isMaxSelected,
+      amountToSwap,
+      amountToReceive,
+      useFlashLoan,
+    }) => {
+      const pool = getCorrectPool();
+      const user = get().account;
+      const chainId = get().currentChainId;
+
+      const { swapCallData, augustus } = await getSwapCallData({
+        srcToken: poolReserve.underlyingAsset,
+        srcDecimals: poolReserve.decimals,
+        destToken: targetReserve.underlyingAsset,
+        destDecimals: targetReserve.decimals,
+        user,
+        route: priceRoute as OptimalRate,
+        chainId,
+      });
+
+      return pool.swapCollateral({
+        fromAsset: poolReserve.underlyingAsset,
+        toAsset: targetReserve.underlyingAsset,
+        swapAll: isMaxSelected,
+        fromAToken: poolReserve.aTokenAddress,
+        fromAmount: amountToSwap,
+        minToAmount: amountToReceive,
+        user,
+        flash: useFlashLoan,
+        augustus,
+        swapCallData,
+      });
+    },
+    setUserEMode: async (categoryId) => {
+      const pool = getCorrectPool() as Pool;
+      const user = get().account;
+      return pool.setUserEMode({
+        user,
+        categoryId,
+      });
+    },
+    signERC20Approval: async (args) => {
+      const pool = getCorrectPool() as Pool;
+      const user = get().account;
+      return pool.signERC20Approval({
+        ...args,
+        user,
+      });
+    },
+    claimRewards: async ({ selectedReward }) => {
+      // TODO: think about moving timestamp from hook to EventEmitter
+      const timestamp = dayjs().unix();
+      const reserves = selectFormattedReserves(get(), timestamp);
+      const currentAccount = get().account;
+
+      const allReserves: string[] = [];
```

Comment:
I'm pretty confident this is wrong.
It's straight from the ui so it's wrong on the interface already but might make sense to fix.

The point is it doesn't really make sense to pass all the assets as we already know which assets have rewards and which don't via `userIncentiveData` but instead we pass every single token of the market (which has incentives).
Am i misreading this?

Reply:
This is correct, there was a change made to the V3 IncentivesController to not store the accrued rewards in a separate variable after withdraw/repay, so passing the entire array to `assets` is the only way to ensure that rewards can be claimed for tokens that are no longer supplied or borrowed. https://github.com/aave/interface/pull/704